purchases/src/main/kotlin/com/revenuecat/purchases/identity/IdentityManager.kt

```diff
@@ -20,14 +22,15 @@ import com.revenuecat.purchases.subscriberattributes.caching.SubscriberAttribute
 import java.util.Locale
 import java.util.UUID
 
-@Suppress("TooManyFunctions")
+@Suppress("TooManyFunctions", "LongParameterList")
 internal class IdentityManager(
     private val deviceCache: DeviceCache,
     private val subscriberAttributesCache: SubscriberAttributesCache,
     private val subscriberAttributesManager: SubscriberAttributesManager,
     private val offeringsCache: OfferingsCache,
     private val backend: Backend,
     private val offlineEntitlementsManager: OfflineEntitlementsManager,
+    private val dispatcher: Dispatcher,
 ) {
 
     val currentAppUserID: String
```

Comment:
This could potentially fail now since setting the user id happens asynchronously now right?

Reply:
hmm right... actually... any call in this class right? for example `currentUserIsAnonymous`. 

I think we need to make all async and make sure they are all enqueued in the same dispatcher, right?

Reply:
Yup pretty much, which seems it would be a pretty big undertaking...